google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionImplTest.java

```diff
@@ -412,6 +412,22 @@ public void testExecuteSetAutocommitOff() {
     }
   }
 
+  @Test
+  public void testExecuteAutocommitNoop() {
+    try (ConnectionImpl subject =
+        createConnection(
+            ConnectionOptions.newBuilder()
+                .setCredentials(NoCredentials.getInstance())
+                .setUri(URI)
+                .build())) {
+      assertThat(subject.isAutocommit(), is(true));
```

Comment:
We should also add tests for the following scenarios:
1. Calling `setAutoCommit(false)` when `autocommit=false` and the transaction has not yet been started (that is; there has been no statement executed on the connection yet).
1. Calling `setAutoCommit(false)` when `autocommit=false` and the transaction has been started (that is; there has been at least one statement executed on the connection).
1. Calling `setAutoCommit(true)` when `autocommit=true`, but the user has first executed `connection.execute(Statement.of("begin transaction"))`

The last test case should also be extended to include calling `setAutoCommit(false)` and be tested both before and after a different statement has been executed in the transaction. The latter means testing all reasonable variations of:

```
set autocommit=true|false;
begin transaction;
[select * from my_table];  -- Include/exclude this statement in the test to force the actual start of the transaction
set autocommit=true|false;
```

Reply:
Added tests for different combinations